### ISSUE TITLE
[AVR] incorrect stack offset of char type arguments passed via stack

### ISSUE THREAD

**Issue by beenshi** [llvm/llvm-project#54148]
For the following C code
```c
char add(char a, char b, ...) {
  return a + b;
}
```

avr-gcc generates
```
add:
	push r28
	push r29
	in r28,__SP_L__
	in r29,__SP_H__
/* prologue: function */
/* frame size = 0 */
/* stack size = 2 */
.L__stack_usage = 2
	ldd r24,Y+6
	ldd r25,Y+5
	add r24,r25
/* epilogue start */
	pop r29
	pop r28
	ret
```
The argument `a` is put on `Y+6` and `b` is put on `Y+5`.

But clang generates
```
add:                                    ; @add
; %bb.0:
	push	r28
	push	r29
	in	r28, 61
	in	r29, 62
	ldd	r24, Y+5
	ldd	r25, Y+7
	add	r25, r24
	mov	r24, r25
	lsl	r25
	sbc	r25, r25
	pop	r29
	pop	r28
	ret
```
The argument `a` is put on `Y+7` and `b` is put on `Y+5`.

A char type argument should only cost 1 byte when passing by stack frame.

**Comment by benshi001** [comment]
This bug is fixed in https://reviews.llvm.org/D121041

**Comment by tstellar** [comment]
This issue was not resolved in time for 14.0.0 and will need to be fixed in the next release (14.0.1).  If you disagree with this or you feel there was a mistake, please change the milestone back to 'LLVM 14.0.0 Release' so we can discuss.

**Comment by benshi001** [comment]
Can anybody help me review it? The code owner of AVR and other AVR contributors do not give me any feedback. Actually this is a simple change. I am quite sure my solution is correct.

**Comment by benshi001** [comment]
fixed in 
https://reviews.llvm.org/D121767

**Comment by benshi001** [comment]
fixed in commit https://github.com/llvm/llvm-project/commit/3fd9a320da8adbefa47071f70667d641f7dd26f2

**Comment by tstellar** [comment]
@aykevl @dylanmckay What do you think about backporting this?

**Comment by tstellar** [comment]
Merged: 5f6fe6b93e85aa90b92681037f4dd42403368bb4
